### ISSUE TITLE
use `str` instead of `repr` for tags in circuit diagram

### ISSUE THREAD

**Issue by richrines1** [quantumlib/Cirq#6411]
**Is your feature request related to a use case or problem? Please describe.**

currently tags are displayed in circuit diagrams via their `repr` ([implicitly](https://github.com/quantumlib/Cirq/blob/4d35768c2a21204d430db5f2b389663e03a9351e/cirq-core/cirq/protocols/circuit_diagram_info_protocol.py#L356-L357) via `str(list(tags))`). This can lead to some unruly or hard-to-read circuit diagrams, e.g.:

```
1: ───X^0.5───Rz(0.928π)[cirq.VirtualTag()]───X^0.5───Rz(π)[cirq.VirtualTag()]───────@──────────────────────────────────
                                                                                     │
2: ───X^0.5───Rz(0.072π)[cirq.VirtualTag()]───X^0.5───Rz(-0.5π)[cirq.VirtualTag()]───X───Rz(-1.5π)[cirq.VirtualTag()]───
```

this could be cleaned up by overriding `tag.__repr__`, but usually not without breaking the expectation that `eval(repr(tagged_op)) == tagged_op`

**Describe the solution you'd like**

it seems like it'd be much more natural (and easier to customize) if the circuit diagram drawer to instead use the tags' strings, e.g.:

```
1: ───X^0.5───Rz(0.928π)[<virtual>]───X^0.5───Rz(π)[<virtual>]───────@──────────────────────────
                                                                     │
2: ───X^0.5───Rz(0.072π)[<virtual>]───X^0.5───Rz(-0.5π)[<virtual>]───X───Rz(-1.5π)[<virtual>]───
```

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

(happy to work on this if it seems reasonable)

**Comment by maffoo** [comment]
Another possibility would be to allow tags to implement some protocol method to produce a more compact representation for circuit diagrams. Something like `_circuit_diagram_info_` itself allows for gates.

**Comment by xXxH3LIOSxXx** [comment]
This is AL from the CirqCync - please assign to me

**Comment by richrines1** [comment]
^ i actually have a working version of this  locally i can push :)

it just uses `__str__` instead of `__repr__` though - if you want to also check for `tag._circuit_diagram_info_` as suggested [above](https://github.com/quantumlib/Cirq/issues/6411#issuecomment-1899384789) that would have to be added (maybe as a separate pr?)

**Comment by pavoljuhas** [comment]
Thank you @richrines1 and @xXxH3LIOSxXx for helping to improve this.
I am assigning to richrines1 as he has already submitted a PR.

**Comment by pavoljuhas** [comment]
csynkque meeting - TODO @pavoljuhas - create a new issue for supporting the `_circuit_diagram_info_` protocol

**Comment by pavoljuhas** [comment]
Continued as #6560.  Anyone interested (@xXxH3LIOSxXx, @richrines1) , please feel free to comment there if you'd like to take it on.  Thank you!
